windows/utils/win_install_online_updates.yml

```diff
@@ -39,17 +51,71 @@
   ignore_errors: "{{ win_updates_ignore_errors | default(false) }}"
 
 - name: "Print Windows online updates install result"
-  debug: var=win_updates_install_result
+  ansible.builtin.debug: var=win_updates_install_result
   when: enable_debug
 
-- name: "Set fact of Windows online updates install status"
-  ansible.builtin.set_fact:
-    win_online_updates_installed: true
+- name: "Windows online updates install task succeeds"
   when:
+    - win_updates_install_result.failed is defined
+    - not win_updates_install_result.failed
     - win_updates_install_result.found_update_count is defined
-    - win_updates_install_result.found_update_count | int != 0
     - win_updates_install_result.installed_update_count is defined
-    - win_updates_install_result.found_update_count | int == win_updates_install_result.installed_update_count | int
+  block:
+    # No updates found, so no updates installed
+    - name: "Set fact of Windows online updates install succeeds"
+      ansible.builtin.set_fact:
+        win_online_updates_succeed: true
+      when: win_updates_install_result.found_update_count | int == 0
+
+    # Updates found, updates installed
+    - name: "Installed update count is equal to found update count"
+      ansible.builtin.set_fact:
+        win_online_updates_succeed: true
+        win_online_updates_installed: true
+      when:
+        - win_updates_install_result.found_update_count | int != 0
+        - win_updates_install_result.installed_update_count | int == win_updates_install_result.found_update_count | int
+
+    # Updates found, while some not installed
+    - name: "Installed update count is not equal to found update count"
+      when:
+        - win_updates_install_result.found_update_count | int != 0
+        - win_updates_install_result.found_update_count | int > win_updates_install_result.installed_update_count | int
+      block:
+        - name: "Set fact of Windows online updates install status"
+          ansible.builtin.set_fact:
+            win_online_updates_installed: true
+          when: win_updates_install_result.installed_update_count | int != 0
+
+        - name: "Set fact of not installed updates list"
+          ansible.builtin.set_fact:
+            win_updates_not_installed: "{{ win_updates_install_result.updates.values() | selectattr('installed', 'equalto', false) }}"
+        - name: "Display the not installed updates list"
+          ansible.builtin.debug: var=win_updates_not_installed
+
+        - name: "Set fact of failure messages of not installed updates"
+          ansible.builtin.set_fact:
+            win_updates_failure_msgs: "{{ win_updates_not_installed | map(attribute='failure_msg') }}"
+
+        - name: "No failure message for update"
```

Comment:
I think the task is not needed. It may make user a litter confused.

Or just keep the first part of the message. The last part seems a little complex to understand.

Reply:
Maybe the debug message can be changed to:
```
 {{ win_updates_not_installed | length - win_updates_failure_msgs | length }} updates failed to be installed without failure messages, which can not be ignored.
```
or something like this?

Reply:
Updated, thanks.